sentry_sdk/integrations/django/asgi.py

```diff
@@ -91,3 +97,52 @@ async def sentry_wrapped_callback(request, *args, **kwargs):
             return await callback(request, *args, **kwargs)
 
     return sentry_wrapped_callback
+
+
+def _asgi_middleware_mixin_factory(_check_middleware_span):
+    # type: (F) -> Any
```

Comment:
```suggestion
    # type: (Callable[..., Any]) -> Any
```

We use typevar if we want to make the return type dependent on the input type